docs/api-reference/index.rst

```diff
@@ -66,3 +66,25 @@ write but in order to avoid duplicated effort, they are defined by HPy.
    :maxdepth: 2
 
    inline-helpers
+
+Adding New API
+==============
+
+1. Add the function to ``hpy/tools/autogen/public_api.h`` and add the mapping
+   from the function to the cpython equivalent to ``hpy/tools/autogen/conf.py``.
```

Comment:
The explicit mapping only needs to be done if the HPy and C API functions do not have the same name (except of the leading `H` in the HPy case). But for description, it is probably easier to put it this way.